### PR TITLE
add actuator test and configuration commands

### PR DESCRIPTION
**Pull request by bkueng**
Split off from https://github.com/mavlink/mavlink/pull/1729

### CODE REVIEW

**Comment by hamishwillee** [comment]
@bkueng FYI I updated the reserved params to use the [syntax recommended in the guide](https://mavlink.io/en/guide/define_xml_element.html#reserved). Technically the guide says you can omit the param altogether if it is reserved with default of 0, but I kind of like that you were explicit. 
The reason to use attributes is we can then generate consistent docs for all cases.

This looks good to me.

**Comment by hamishwillee** [comment]
@auturgy This is the actuator test and actuator config messages that have been split out of the PR for the actuator component information: https://github.com/mavlink/mavlink/pull/1729

As we discussed this is a much better message than [MAV_CMD_DO_MOTOR_TEST](https://mavlink.io/en/messages/common.html#MAV_CMD_DO_MOTOR_TEST) and therefore likely to be something that ArduPilot would choose to update to. As this has been prototyped this therefore PRs into common.xml.

You OK to approve?

**Comment by bkueng** [comment]
> @bkueng FYI I updated the reserved params to use the syntax recommended in the guide.

Thanks

**Comment by auturgy** [comment]
lgtm

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -817,6 +817,129 @@
...
     </enum>
+    <enum name="ACTUATOR_CONFIGURATION">
+      <description>Actuator configuration, used to change a setting on an actuator. Component information metadata can be used to know which outputs support which commands.</description>
+      <entry value="0" name="ACTUATOR_CONFIGURATION_NONE">
+        <description>Do nothing.</description>
+      </entry>
+      <entry value="1" name="ACTUATOR_CONFIGURATION_BEEP">
+        <description>Command the actuator to beep now.</description>
+      </entry>
+      <entry value="2" name="ACTUATOR_CONFIGURATION_3D_MODE_ON">
```
[the remark this answers is not in the document]
Comment (julianoes) [reply message_definitions/v1.0/common.xml:828]: Given there is also a spin reverse option, this might be confusing, os 3D mode is probably ok.

**Review by julianoes — APPROVED** [review]

File: message_definitions/v1.0/common.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hamishwillee) [reply message_definitions/v1.0/common.xml:828]: @julianoes I'm not sure whether you're agreeing with your suggestion or disagreeing.

@bkueng What's the difference between changing the spin direction and reversing the thrust (3D mode)?

Further, why does it make sense that the spin direction is not fixed to CW or CCW?

**Comment by hamishwillee** [comment]
As discussed in the [20211110](https://github.com/mavlink/mavlink/wiki/20211110-Dev-Meeting) dev call
- The question was asked as to why have motor and servo functions - why not make these completely abstract and map them "under the hood" (since now you'll have to add more functions for other things like parachute, camera etc). The answer given was that this makes it easier for GCS and flight stacks - they tend to share an understanding that a particular motor is numbered - and this makes using the commands easier.
- "Are there enough motors"? Answered with "gaps have been left to allow more to be added if needed.

In general we're happy and this is approved. Action on Julian and Beat to finalise the naming/wording of the enums, in particular around "3D on/off", but after that this can be approved.

File: message_definitions/v1.0/common.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bkueng) [reply message_definitions/v1.0/common.xml:828]: > @bkueng What's the difference between changing the spin direction and reversing the thrust (3D mode)?

That's the main reason I wanted to have as distinct names as possible. 3D mode maps half of the setpoint range to CW, the other to CCW, whereas without the full range is either CW or CCW.

> Further, why does it make sense that the spin direction is not fixed to CW or CCW?

There's no way of knowing which direction is CW vs CCW.

[the remark this answers is not in the document]
Comment (julianoes) [reply message_definitions/v1.0/common.xml:828]: Ok makes sense.

**Pull request merged**